src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -659,6 +666,80 @@ static bool ConflictsWith(EnumFieldInfo current, EnumFieldInfo other)
             }
         }
 
+        /// <summary>
+        /// Performs a topological sort on enum fields using Kahn's algorithm to ensure values that are supersets of other values come first.
+        /// </summary>
+        private static EnumFieldInfo[] TopologicalSortEnumFields(EnumFieldInfo[] enumFields)
```

Comment:
Is this topsort how `Enum.ToString()` also does it?

The current algorithm is `O(n^2)` because of the adjacency list creation step. A simpler alternative to the current ordering would be to order the enum fields by the number of bits each of them has set. So 0b111 comes before 0b011 since they have 3 and 2 bits set respectively. This would be `O(n)` if you have `sizeof(long)` buckets and just put each enum into the bucket representing its bit count. And the bonus is that this ordering is also a topsort.

Reply:
Good idea, I'll change it to that.

Reply:
Done.